src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/NoWhitespaceBeforeCaseDefaultColonCheck.java

```diff
@@ -0,0 +1,227 @@
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+////////////////////////////////////////////////////////////////////////////////
+
+package com.puppycrawl.tools.checkstyle.checks.whitespace;
+
+import com.puppycrawl.tools.checkstyle.StatelessCheck;
+import com.puppycrawl.tools.checkstyle.api.AbstractCheck;
+import com.puppycrawl.tools.checkstyle.api.DetailAST;
+import com.puppycrawl.tools.checkstyle.api.TokenTypes;
+import com.puppycrawl.tools.checkstyle.utils.TokenUtil;
+
+/**
+ * <p>
+ * Checks that there is no whitespace before the colon in a switch block.
+ * </p>
+ * <p>
+ * To configure the check:
+ * </p>
+ * <pre>
+ * &lt;module name=&quot;NoWhitespaceBeforeCaseDefaultColon&quot;/&gt;
+ * </pre>
+ * <p>Example:</p>
+ * <pre>
+ * class Test {
+ *   {
+ *     switch(1) {
+ *         case 1 : // violation, whitespace before ':' is not allowed here
+ *             break;
+ *         case 2: // OK
+ *             break;
+ *         default : // violation, whitespace before ':' is not allowed here
+ *             break;
+ *     }
+ *
+ *     switch(2) {
+ *         case 2: // OK
+ *             break;
+ *         case 3, 4
+ *                  : break; // violation, ':' should not be in separate line
+ *         case 4,
+ *               5: break; // ok
+ *         default
+ *               : // violation, ':' should not be in separate line
+ *             break;
+ *     }
+ *
+ *     switch(day) {
+ *         case MONDAY, FRIDAY, SUNDAY: System.out.println("  6"); break;
+ *         case TUESDAY               : System.out.println("  7"); break; // violation
+ *   }
+ * }
+ * </pre>
+ * <p>
+ * Parent is {@code com.puppycrawl.tools.checkstyle.TreeWalker}
+ * </p>
+ * <p>
+ * Violation Message Keys:
+ * </p>
+ * <ul>
+ * <li>
+ * {@code ws.colonInSeparateLine}
+ * </li>
+ * <li>
+ * {@code ws.preceded}
+ * </li>
+ * </ul>
+ *
+ * @since 8.43
+ */
+@StatelessCheck
+public class NoWhitespaceBeforeCaseDefaultColonCheck
+    extends AbstractCheck {
+
+    /**
+     * A key is pointing to the warning message text in "messages.properties"
+     * file.
+     */
+    public static final String MSG_KEY = "ws.preceded";
+
+    /**
+     * A key is pointing to the warning message text in "messages.properties"
+     * file.
+     */
+    public static final String MSG_KEY_ON_SEPARATE = "ws.colonInSeparateLine";
+
+    @Override
+    public int[] getDefaultTokens() {
+        return getRequiredTokens();
+    }
+
+    @Override
+    public int[] getAcceptableTokens() {
+        return getRequiredTokens();
+    }
+
+    @Override
+    public int[] getRequiredTokens() {
+        return new int[] {TokenTypes.COLON};
+    }
+
+    @Override
+    public boolean isCommentNodesRequired() {
+        return true;
+    }
+
+    @Override
+    public void visitToken(DetailAST ast) {
+        if (isInSwitch(ast)) {
+            if (isColonOnDifferentLine(ast)) {
```

Comment:
Please point me discussion where we made decision  to have separate message on this.
Separate line already mean there is space before `:`

Reply:
> Please point me discussion where we made decision to have separate message on this.

We have not discussed it.

> Separate line already mean there is space before `:`

True, but this statement becomes confusing for some cases like 
```
case 1
      :  break;
```
and 
```
case 1
: break;
```

It both cases if we give message `: is preceded with whitespace.`, doesn't gives a clear picture.

Reply:
We need to compare line and column.

>It both cases if we give message : is preceded with whitespace., doesn't gives a clear picture.

Hm, for me it is clear. 
There might be tab symbol in between (not a whitespace) and this might be confusing a bit. But can change wording a bit to be general on any non printable symbol in between.

Reply:
> We need to compare line and column.

I will do change.

> There might be tab symbol in between (not a whitespace) and this might be confusing a bit. But can change wording a bit to be general on any non printable symbol in between.

What do think of a single message like : `''{0}'' is preceded with non printable symbol.` for both cases?

Reply:
Is the name of the check and its definition is required to be changed? 
I mean should we change name from `NoWhitespaceBeforeCaseDefaultColonCheck` ->`NoNonPrintableSymbolBeforeCaseDefaultColonCheck`
and the definition of check to 
`Checks that there is no whitespace before the colon in a switch block.` -> `Checks that there is no non printable symbol before the colon in a switch block.`

Reply:
>is preceded with whitespace

Let's use this, us us simple and good enough.
And keep Check name.

Reply:
Done.